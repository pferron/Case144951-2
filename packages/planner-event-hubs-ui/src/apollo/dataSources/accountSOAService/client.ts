import { LoggerFactory } from '@cvent/logging/LoggerFactory';
import { AccountLocale, AccountVideoCenterConfig, AccountDetails } from '@cvent/planner-event-hubs-model/types';
import { CvestDataSource } from '@dataSources/CvestDataSource';
import { normalizeRawOrJSONResponseToJSON } from '@server/utils';

const LOG = LoggerFactory.create('account-soa-client');

export class AccountSOAClient extends CvestDataSource {
  constructor() {
    super();
    this.baseURL = `${process.env.ACCOUNT_SOA_SERVICE}/v1`;
  }

  /**
   * Fetches AccountConfig from Account SOA. Returns very small set
   * of AccountConfig data, just what's needed by VideoCenter today.
   *
   * @returns AccountConfig
   */
  public async getAccountConfig(): Promise<AccountVideoCenterConfig> {
    LOG.debug('getAccountConfig()');

    // RED
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const response = normalizeRawOrJSONResponseToJSON(await this.get('/accountconfig')) as any;

    return {
      AccountFeatures: {
        GeneralFeatures: {
          AIWritingAssistantEnabled: response?.AccountFeatures?.GeneralFeatures?.AIWritingAssistantEnabled,
          AllowCodeSnippets: response?.AccountFeatures?.GeneralFeatures?.AllowCodeSnippets,
          AllowCustomFonts: response?.AccountFeatures?.GeneralFeatures?.AllowCustomFonts,
          AllowGoogleAnalytics: response?.AccountFeatures?.GeneralFeatures?.AllowGoogleAnalytics,
          AllowOAuth: response?.AccountFeatures?.GeneralFeatures?.AllowOAuth
        },
        Blades: {
          AllowVideosBlade: response?.AccountFeatures?.Blades?.AllowVideosBlade
        }
      },
      VideoManagementFeatures: {
        VideoStorageSize: response?.VideoManagementFeatures?.VideoStorageSize,
        VideoCenterFlag: response?.VideoManagementFeatures?.VideoCenterFlag
      },
      InternationalSettings: {
        DefaultLanguageId: response?.AccountFeatures?.InternationalSettings?.DefaultLanguageId,
        DefaultCultureCode: response?.AccountFeatures?.InternationalSettings?.DefaultCultureCode
      },
      EventFeatures: {
        GeneralFeatures: {
          LicenseTypeId: response?.EventFeatures?.GeneralFeatures?.LicenseTypeId
        }
      }
    };
  }

  /**
   * Fetches AccountLocale from Account SOA.
   *
   * @returns AccountLocale
   */
  public async getAccountLocale(): Promise<AccountLocale[]> {
    LOG.debug('getAccountLocale()');

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return normalizeRawOrJSONResponseToJSON(await this.get('/accountlocale')) as any;
  }

  /**
   * Fetches AccountDetails from Account SOA.
   *
   * @returns AccountLocale
   */
  public async getAccountDetails(): Promise<AccountDetails> {
    LOG.debug('getAccountDetails()');

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return normalizeRawOrJSONResponseToJSON(await this.get('/accountdetailextended')) as any;
  }
}
