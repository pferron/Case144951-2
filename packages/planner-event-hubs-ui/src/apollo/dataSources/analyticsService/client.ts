import { CvestDataSource } from '@dataSources/CvestDataSource';
import { LoggerFactory } from '@cvent/auth-client';

const APPLICATION_NAME = 'Planner Event Hubs';

const LOG = LoggerFactory.create('analytics-api-client');
export interface AnalyticsField {
  fieldKey: string;
  fieldType: string;
  encrypt: boolean;
}

export interface AnalyticsRowData {
  values: Array<string>;
}

export interface VideoViewsAndPosition {
  videoId: string;
  totalViews: number;
  position: number;
}

export interface AnalyticsResponseData {
  total: number;
  fields: Array<AnalyticsField>;
  rows: Array<AnalyticsRowData>;
}

export class AnalyticsClient extends CvestDataSource {
  constructor() {
    super();
    this.baseURL = `${process.env.ANALYTICS_DATASOURCE_URL}`;
  }

  getTotalViewsByHubId = async (
    hubId: string,
    startDate: string,
    endDate: string,
    pipeline?: string
  ): Promise<AnalyticsResponseData> => {
    LOG.debug('getTotalViewsByHubId()', hubId);

    const params = [
      {
        name: 'startTime',
        value: startDate
      },
      {
        name: 'endTime',
        value: endDate
      }
    ];

    const requestBody = {
      context: {
        application: APPLICATION_NAME
      },
      criteria: {
        columns: ['time_window_per_day', 'entity_id', 'total_views'],
        expression: `entity_id = '${hubId}'`,
        params
      }
    };

    const dataSource = pipeline === 'v2' ? 'total_views_per_hub_per_day_agg_v2' : 'total_views_per_hub_per_day_agg';
    const data = (await this.post(`/v1/analytics_data_source/requestData/${dataSource}`, requestBody, {
      headers: { authorization: `API_KEY ${process.env.API_KEY}` }
    })) as AnalyticsResponseData;

    return data;
  };

  averageViewDurationByHubId = async (
    hubId: string,
    startDate: string,
    endDate: string,
    pipeline?: string
  ): Promise<AnalyticsResponseData> => {
    LOG.debug('averageViewDurationByHubId()', hubId);

    const params = [
      {
        name: 'startTime',
        value: startDate
      },
      {
        name: 'endTime',
        value: endDate
      }
    ];

    const requestBody = {
      context: {
        application: APPLICATION_NAME
      },
      criteria: {
        columns: ['time_window_per_day', 'entity_id', 'average_view_duration'],
        expression: `entity_id = '${hubId}'`,
        params
      }
    };

    const dataSource =
      pipeline === 'v2' ? 'average_view_duration_per_hub_per_day_agg_v2' : 'average_view_duration_per_hub_per_day_agg';
    const data = (await this.post(`/v1/analytics_data_source/requestData/${dataSource}`, requestBody, {
      headers: { authorization: `API_KEY ${process.env.API_KEY}` }
    })) as AnalyticsResponseData;

    return data;
  };

  getTotalVideosViewedByHubId = async (
    mappingId: string,
    hubId: string,
    startDate: string,
    endDate: string,
    pipeline?: string
  ): Promise<AnalyticsResponseData> => {
    LOG.debug('getTotalVideosViewedByHubId()', hubId, startDate, endDate);
    const params = [
      {
        name: 'mapping_id',
        value: mappingId
      },
      {
        name: 'startTime',
        value: startDate
      },
      {
        name: 'endTime',
        value: endDate
      }
    ];

    const requestBody = {
      context: {
        application: APPLICATION_NAME
      },
      criteria: {
        columns: ['time_window_per_day', 'video_id', 'entity_id', 'total_views'],
        expression: `entity_id = '${hubId}' AND view_type='Recording'`,
        params
      }
    };

    const dataSource =
      pipeline === 'v2' ? 'total_views_per_source_per_day_agg_v2' : 'total_views_per_source_per_day_agg';
    const data = await this.post(`/v1/analytics_data_source/requestData/${dataSource}`, requestBody, {
      headers: { authorization: `API_KEY ${process.env.API_KEY}` }
    });
    return data;
  };

  viewsByDeviceType = async (
    hubId: string,
    startDate: string,
    endDate: string,
    pipeline?: string
  ): Promise<AnalyticsResponseData> => {
    LOG.debug('viewsByDeviceType()', hubId);

    const params = [
      {
        name: 'startTime',
        value: startDate
      },
      {
        name: 'endTime',
        value: endDate
      }
    ];

    const requestBody = {
      context: {
        application: APPLICATION_NAME
      },
      criteria: {
        columns: ['time_window_per_day', 'entity_id', 'device_type', 'total_views'],
        expression: `entity_id = '${hubId}'`,
        params
      }
    };

    const dataSource =
      pipeline === 'v2'
        ? 'total_views_by_device_type_per_hub_per_day_agg_v2'
        : 'total_views_by_device_type_per_hub_per_day_agg';
    const data = (await this.post(`/v1/analytics_data_source/requestData/${dataSource}`, requestBody, {
      headers: { authorization: `API_KEY ${process.env.API_KEY}` }
    })) as AnalyticsResponseData;

    return data;
  };

  getAllVideosViewedByHubId = async (
    mappingId: string,
    hubId: string,
    startDate: string,
    endDate: string,
    limit: number,
    offset: number
  ): Promise<AnalyticsResponseData> => {
    LOG.debug('getAllVideosViewedByHubId()', hubId, startDate, endDate);
    const params = [
      {
        name: 'mapping_id',
        value: mappingId
      },
      {
        name: 'startTime',
        value: startDate
      },
      {
        name: 'endTime',
        value: endDate
      },
      {
        name: 'limit',
        value: limit
      },
      {
        name: 'offset',
        value: offset
      }
    ];

    const requestBody = {
      context: {
        application: APPLICATION_NAME
      },
      criteria: {
        columns: ['video_id', 'total_views'],
        expression: `entity_id = '${hubId}'`,
        params
      }
    };

    return this.post(`/v1/analytics_data_source/requestData/total_views_per_hub_per_video_agg_v2`, requestBody, {
      headers: { authorization: `API_KEY ${process.env.API_KEY}` }
    });
  };

  getMemberVideoWatchDurationByHubId = async (
    mappingId: string,
    hubId: string,
    videoId: string,
    startDate: string,
    endDate: string,
    limit: number,
    offset: number
  ): Promise<AnalyticsResponseData> => {
    LOG.debug('getMemberVideoWatchDurationByHubId()', hubId, videoId, startDate, endDate);
    const params = [
      {
        name: 'mapping_id',
        value: mappingId
      },
      {
        name: 'startTime',
        value: startDate
      },
      {
        name: 'endTime',
        value: endDate
      },
      {
        name: 'limit',
        value: limit
      },
      {
        name: 'offset',
        value: offset
      }
    ];

    const requestBody = {
      context: {
        application: APPLICATION_NAME
      },
      criteria: {
        columns: ['unique_id', 'total_duration'],
        expression: `entity_id = '${hubId}' AND video_id = '${videoId}'`,
        params
      }
    };

    return this.post(`/v1/analytics_data_source/requestData/total_watch_duration_per_member_agg_v2`, requestBody, {
      headers: { authorization: `API_KEY ${process.env.API_KEY}` }
    });
  };
}
