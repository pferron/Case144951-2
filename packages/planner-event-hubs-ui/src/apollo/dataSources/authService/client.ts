import { LoggerFactory } from '@cvent/logging/LoggerFactory';
import { GrantedAccessToken } from '@cvent/auth-client';
import { AUTH_COOKIE_MAX_AGE } from '@utils/constants';
import { RequestOptions, RESTDataSource } from '../RESTDataSource/RESTDataSource';

const LOG = LoggerFactory.create('auth-api');

/**
 * This class does not extend CvestDataSource because we don't want to set all the same headers and params from here.
 * The intent of this class is to provide a path to generate a bearer for public/anonymous users when visiting a public Video Center.
 *
 * This class also provides a clear and concise example of a DataSource that does not use CvestDataSource.
 *
 * If further work is needed here, consider removing this class and using @cvent/auth-client instead.
 */
export class AuthClient extends RESTDataSource {
  constructor() {
    super();
    this.baseURL = `${process.env.AUTH_SERVICE}/v1`;
  }

  public willSendRequest(request: RequestOptions): void {
    this.setStandardHeaders(request);
  }

  public setStandardHeaders(request: RequestOptions): void {
    request.headers.set('Authorization', `API_KEY ${process.env.API_KEY}`);
  }

  // TODO: remove publicLogin from authService
  public async publicLogin(
    hubId: string,
    environment: string,
    accountMappingId: string,
    accountId: string
  ): Promise<GrantedAccessToken> {
    LOG.debug('publicLogin sending metadata', { accountMappingId, hubId, environment, accountId });
    return this.post(`/access_token?accessTokenTTL=${AUTH_COOKIE_MAX_AGE}`, {
      metadata: {
        accountMappingId,
        VideoCenterId: hubId,
        environment,
        accountId,
        AccountId: accountId
      },
      grantedAuthorizations: [
        {
          appId: process.env.VIDEO_HUB_SERVICE_APP_ID,
          roles: ['video-center:anonymous']
        },
        {
          appId: process.env.UNIVERSAL_VIDEO_APP_ID,
          roles: ['videos:read']
        },
        {
          appId: process.env.UNIVERSAL_CONTACTS_APP_ID,
          roles: ['contacts:read']
        }
      ]
    });
  }

  public async verifyToken(accessToken: string): Promise<GrantedAccessToken> {
    LOG.debug('verifying Token');
    return this.get(`/access_token/${accessToken}/verify`);
  }

  public async accessTokenRefresh(accessToken: string): Promise<void> {
    LOG.debug('accessToken Refreshing');
    return this.put(`/access_token/${accessToken}/refresh`);
  }
}
