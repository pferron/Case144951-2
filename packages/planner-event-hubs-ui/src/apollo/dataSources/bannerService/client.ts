import { LoggerFactory } from '@cvent/logging/LoggerFactory';
import { CvestDataSource } from '@dataSources/CvestDataSource';
import {
  NewBanner,
  BannerPagingResponse,
  ExistingBanner,
  FilterInput,
  BannerUpdate
} from '@cvent/planner-event-hubs-model/types';

const LOG = LoggerFactory.create('banner-service-client');
export class BannerServiceClient extends CvestDataSource {
  constructor() {
    super();
    this.baseURL = `${process.env.VIDEO_HUB_SERVICE}/v1/`;
  }

  getBanner = async (centerId: string, bannerId: string): Promise<ExistingBanner> => {
    LOG.debug('getBanner(bannerId)', bannerId);
    return this.get(`video-hubs/${centerId}/banners/${bannerId}`);
  };

  updateBanner = async (bannerUpdate: BannerUpdate): Promise<ExistingBanner> => {
    LOG.debug('updateBanner(bannerId)', bannerUpdate);
    return this.put(`video-hubs/${bannerUpdate.centerId}/banners/${bannerUpdate.id}`, bannerUpdate, {
      cacheRefreshOptions: { update: true }
    });
  };

  getBanners = async (centerId: string, filterInput: FilterInput): Promise<BannerPagingResponse> => {
    LOG.debug('getBanners(filterInput)', filterInput);
    const params: { token?: string; limit?: number } = {};

    if (filterInput?.token) {
      params.token = filterInput.token;
    }

    if (filterInput?.limit) {
      params.limit = filterInput.limit;
    }
    return this.get(`video-hubs/${centerId}/banners`, params);
  };

  bannerCreate = async (banner: NewBanner): Promise<ExistingBanner> => {
    LOG.debug('bannerCreate(banner)', banner);
    return this.post(`video-hubs/${banner.centerId}/banners`, banner);
  };

  bannerDelete = async (centerId: string, bannerId: string): Promise<void> => {
    LOG.debug('bannerDelete(bannerId)', bannerId);
    return this.delete(`video-hubs/${centerId}/banners/${bannerId}`, undefined, {
      cacheRefreshOptions: { evict: true }
    });
  };

  getBannersByIds = async (centerId: string, bannerIds: string[]): Promise<ExistingBanner[]> => {
    LOG.debug(`getBannersByIds (${bannerIds.length})`);

    return Promise.all(
      bannerIds.map(async (id): Promise<ExistingBanner> => {
        return this.getBanner(centerId, id);
      })
    );
  };
}
