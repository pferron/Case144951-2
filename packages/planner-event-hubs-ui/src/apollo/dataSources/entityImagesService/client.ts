import { CvestDataSource } from '@dataSources/CvestDataSource';
import { EntityImageInput, EntityInput } from '@cvent/planner-event-hubs-model/types';
import { NOT_FOUND_ERROR_CODE } from '@utils/constants';
import { getAccountMappingId } from '@resolvers/common/utils/authMetadataUtils';

export interface ImageResponse {
  id: string;
  entityId: string;
  entityType: string;
  name: string;
  relativePath: string;
  optimizedUrl: string;
  createdAt: string;
  filename: string;
  size: number;
  isPrimary: boolean;
}

export interface Paging {
  limit: number;
  currentToken: string;
}

export interface ImageFilterResponse {
  paging?: Paging;
  data: ImageResponse[];
}

export class EntityImagesServiceClient extends CvestDataSource {
  constructor() {
    super();
    this.baseURL = `${process.env.EVENT_IMAGES_SERVICE_URL}/v1/`;
  }

  createEntityFilter = (entity: EntityInput): string => {
    return `primary eq 'true' and entity.type eq '${entity.type}' and entity.id eq '${entity.id}'`;
  };

  getEntityImage = async (entity): Promise<ImageFilterResponse> => {
    try {
      return await this.get(
        `images?filter=${this.createEntityFilter(entity)}`,
        {},
        {
          headers: {
            authorization: `API_KEY ${process.env.API_KEY}`,
            'cvent-account-mapping-id': getAccountMappingId(this.context.auth)
          }
        }
      );
    } catch (e) {
      // We don't want to throw error if no image is added to a channel
      if (e.code === NOT_FOUND_ERROR_CODE) {
        return null;
      }
      throw e;
    }
  };

  deleteEntityImage = async (imageId: string): Promise<void> => {
    return this.delete(`images/${imageId}`);
  };

  uploadEntityImage = async (imageInput: EntityImageInput): Promise<ImageResponse> => {
    return this.post(`images`, imageInput);
  };
}
