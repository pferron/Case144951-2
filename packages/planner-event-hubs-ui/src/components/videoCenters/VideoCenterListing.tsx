import { useMutation, useQuery } from '@apollo/client';
import Header from '@components/Header';
import VideoCenterCreateModal from '@components/videoCenters/create/VideoCenterCreateModal';
import ViewVideoCenters from '@components/videoCenters/listing/ViewVideoCenters';
import ScrollViewWithBars from '@cvent/carina/components/ScrollViewWithBars/ScrollViewWithBars';
import { ActionType } from '@cvent/carina/components/Templates/utils/helpers';
import { Hub, HubsPagingResponse, PagingResponse } from '@cvent/planner-event-hubs-model/types';
import { useDefaultLanguageSetting } from '@hooks/DefaultLanguageProvider';
import useInfiniteScroll from '@hooks/useInfiniteScroll';
import { VIDEO_CENTER_LIST_LIMIT } from '@utils/constants';
import { GET_VIDEO_HUBS, DELETE_VIDEO_HUB } from '@cvent/planner-event-hubs-model/operations/hub';
import { LoggerFactory } from '@cvent/logging/LoggerFactory';
import LoadingWrapper from '@components/common/loading/LoadingWrapper';
import { useTranslate } from 'nucleus-text';
import { useCallback, useRef, useState } from 'react';

const LOG = LoggerFactory.create('Events-plus-listing');

function VideoCenterListing(): JSX.Element {
  const { translate } = useTranslate();
  const videoCenterTableRef = useRef(null);
  const [isCreateModalOpen, setIsCreateModalOpen] = useState<boolean>(false);
  const [videoCentersData, setVideoCentersData] = useState<Array<Hub>>([]);
  const [videoCentersPagination, setVideoCentersPagination] = useState<PagingResponse>(null);
  const [isLoadingMore, setIsLoadingMore] = useState(false);
  const [initialLoadingComplete, setInitialLoadingComplete] = useState(false);

  const {
    fetchMore,
    refetch: refetchVideoCenters,
    error
  } = useQuery(GET_VIDEO_HUBS, {
    fetchPolicy: 'network-only',
    nextFetchPolicy: 'cache-first',
    variables: {
      input: { limit: VIDEO_CENTER_LIST_LIMIT }
    },
    onCompleted: ({ hubs }: { hubs: HubsPagingResponse }) => {
      setVideoCentersData(hubs.data);
      setVideoCentersPagination(hubs.paging);
      setInitialLoadingComplete(true);
    },
    onError: apolloError => {
      setInitialLoadingComplete(true);
      LOG.error(apolloError, 'Failed to get event plus hubs');
    }
  });

  const defaultLang = useDefaultLanguageSetting();

  const loadMore = useCallback(async () => {
    const nextToken = videoCentersPagination?.nextToken;
    if (!isLoadingMore && nextToken) {
      setIsLoadingMore(true);
      await fetchMore({
        variables: {
          input: {
            token: nextToken
          }
        },
        updateQuery: (prev, { fetchMoreResult }: { fetchMoreResult? }) => {
          if (!fetchMoreResult?.hubs?.data) return prev;
          setVideoCentersData(existingCenterList => [...existingCenterList, ...fetchMoreResult.hubs.data]);
          setVideoCentersPagination(fetchMoreResult.hubs.paging);

          return {
            hubs: {
              ...fetchMoreResult.hubs,
              data: [...prev.hubs.data, ...fetchMoreResult.hubs.data]
            }
          };
        }
      });
      setIsLoadingMore(false);
    }
  }, [videoCentersPagination, isLoadingMore, fetchMore]);

  useInfiniteScroll(loadMore, videoCenterTableRef, {
    bottomScrollLeft: 100,
    immediate: true
  });

  const [deleteVideoHubMutation] = useMutation(DELETE_VIDEO_HUB);
  const handleDeleteVideoCenter = useCallback(
    (videoCenterId: string) => {
      deleteVideoHubMutation({
        variables: { input: { id: videoCenterId } },
        onCompleted: () => {
          setVideoCentersData(existingList => existingList.filter(hub => hub.id !== videoCenterId));
          // setIsVideoCenterDeleteSuccess(true); // TODO: add alert
        },
        onError: () => {
          // setIsVideoCenterDeleteSuccess(false); // TODO: add alert
        }
      });
    },
    [deleteVideoHubMutation]
  );

  const headerActions: ActionType[] = [
    {
      value: translate('video_hubs_header_action_create_text'),
      onClick: (): void => setIsCreateModalOpen(true),
      label: translate('video_hubs_header_action_create_text')
    }
  ];

  const videoCenterListingHeader: JSX.Element = (
    <Header title={translate('video_hubs_header_title')} actions={headerActions} />
  );

  const eventsPlusListingRenderer = (): JSX.Element => (
    <>
      <ViewVideoCenters
        videoHubs={videoCentersData}
        loadingVideoCentersList={!initialLoadingComplete && videoCentersData.length === 0}
        headerActions={headerActions}
        deleteVideoCenter={handleDeleteVideoCenter}
      />
      <VideoCenterCreateModal
        isModalOpen={isCreateModalOpen}
        setIsModalOpen={setIsCreateModalOpen}
        defaultLanguage={defaultLang?.DefaultCultureCode}
        refetchVideoCenters={refetchVideoCenters}
      />
    </>
  );

  return (
    <ScrollViewWithBars
      forwardScrollViewRef={videoCenterTableRef}
      header={videoCenterListingHeader}
      css={{ zIndex: 1 }}
    >
      <LoadingWrapper loading={!initialLoadingComplete} renderer={eventsPlusListingRenderer} errors={[error]} />
    </ScrollViewWithBars>
  );
}

export default VideoCenterListing;
