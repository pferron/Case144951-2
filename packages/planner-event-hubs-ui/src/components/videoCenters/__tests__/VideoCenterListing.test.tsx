import React from 'react';
import { render } from '@testing-library/react';
import { TestWrapper } from '@utils/testUtil/TestWrapper';
import VideoCenterListing from '@components/videoCenters/VideoCenterListing';
import { MockedProvider } from '@apollo/client/testing';
import { GET_VIDEO_HUBS } from '@cvent/planner-event-hubs-model/operations/hub';
import 'jest-axe/extend-expect';
import { axe } from 'jest-axe';
import { screen } from 'nucleus-text/testing-library/screen';

interface RouterProps {
  pathname: string;
  route: string;
  query: {
    isSuccess: boolean;
  };
}

jest.mock('next/router', () => ({
  useRouter(): RouterProps {
    return {
      pathname: '/path',
      route: '/route',
      query: { isSuccess: true }
    };
  }
}));

describe('Video Centers', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Render Video Centers empty screen', async () => {
    const mocks = [
      {
        request: {
          query: GET_VIDEO_HUBS,
          variables: {
            input: { limit: 20 }
          }
        },
        result: {
          data: {
            hubs: {
              data: []
            }
          }
        }
      }
    ];

    const { container } = render(
      <TestWrapper>
        <MockedProvider mocks={mocks}>
          <VideoCenterListing />
        </MockedProvider>
      </TestWrapper>
    );
    expect(container).toMatchSnapshot();
    expect(await screen.findByText(/You haven't created any hubs yet/)).toBeInTheDocument();
    expect(screen.getByTestId('video-hub-empty-page-container')).toBeInTheDocument();
    expect(await axe(container)).toHaveNoViolations();
  });

  it('Should render the table with event hubs', async () => {
    const mocks = [
      {
        request: {
          query: GET_VIDEO_HUBS,
          variables: {
            input: { limit: 20 }
          }
        },
        result: {
          data: {
            hubs: {
              data: [
                {
                  __typename: 'Hub',
                  id: 'e68c7b34-4ed8-4c3b-a5b6-20f018048d0f',
                  status: 'Inactive',
                  config: {
                    __typename: 'Config',
                    title: 'New Events Plus',
                    ownerFirstName: '2',
                    ownerLastName: 'e',
                    ownerEmail: 'BC@gmail.com',
                    url: 'https://staging.cvent.me/7zaZ39',
                    locale: 'en-US'
                  }
                }
              ],
              paging: {
                nextToken: null
              }
            }
          }
        }
      }
    ];
    render(
      <TestWrapper>
        <MockedProvider mocks={mocks}>
          <VideoCenterListing />
        </MockedProvider>
      </TestWrapper>
    );
    expect(await screen.findByRole('button', { key: 'video_hubs_header_action_create_text' })).toBeInTheDocument();
    expect(await screen.findByRole('button', { name: 'Row actions' })).toBeInTheDocument();
    expect(screen.getByText('New Events Plus')).toBeInTheDocument();
    /* Axe test fails when Table actions are added, it's from carina Table component which we don't have control */
  });
});
