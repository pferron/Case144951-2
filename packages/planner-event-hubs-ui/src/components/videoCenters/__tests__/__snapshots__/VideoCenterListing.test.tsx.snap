// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Centers Render Video Centers empty screen 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-height: inherit;
  max-width: inherit;
  height: 100%;
  width: 100%;
  overflow: hidden;
  box-sizing: border-box;
  z-index: 1;
}

.emotion-1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  max-width: 100vw;
  max-height: 100vh;
  height: 100%;
  width: 100%;
  overflow-y: auto;
  position: relative;
}

.emotion-2 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  overflow: visible;
  -webkit-transition: box-shadow .2s ease-in-out;
  transition: box-shadow .2s ease-in-out;
}

.emotion-3 {
  position: relative;
}

.emotion-4 {
  direction: ltr;
  background: rgb(255,255,255);
  padding: 24px;
  padding-bottom: 0;
  border-bottom: 1px solid rgb(217,220,222);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-family: 'Rubik','Helvetica','Arial';
  font-weight: 400;
}

.emotion-4 header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  width: 100%;
  margin: 16px 0 32px 0;
  position: relative;
}

@media (max-width: 560px) {
  .emotion-4 header {
    margin: 16px 0 0 0;
  }
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (max-width: 560px) {
  .emotion-5 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: start;
    -webkit-box-align: start;
    -ms-flex-align: start;
    align-items: start;
  }
}

.emotion-6 {
  padding: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0;
}

@media (max-width: 560px) {
  .emotion-6 {
    margin: 16px 0 32px 0;
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  word-break: break-all;
}

.emotion-8 {
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  margin: 0;
}

@media (max-width: 1080px) {
  .emotion-9 {
    position: absolute;
    top: -33px;
    right: 0;
  }
}

.emotion-10 {
  direction: ltr;
  -webkit-flex: 0 0 30%;
  -ms-flex: 0 0 30%;
  flex: 0 0 30%;
  max-height: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-10>*,
.emotion-10>button {
  margin: 0.5rem;
}

.emotion-11 {
  direction: ltr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(0,106,225);
  border: 1px solid transparent;
  border-radius: 10px;
  box-sizing: border-box;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Rubik','Helvetica','Arial';
  font-style: normal;
  font-weight: 400;
  height: 40px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  letter-spacing: 0;
  margin-right: 0;
  margin-left: 0;
  max-width: 100%;
  padding-bottom: 0;
  padding-top: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background-color cubic-bezier(0, 0, 0.39, 1) 100ms,color cubic-bezier(0, 0, 0.39, 1) 100ms,border-color cubic-bezier(0, 0, 0.39, 1) 100ms,box-shadow cubic-bezier(0.2, 0, 0.4, 0.9) 100ms;
  transition: background-color cubic-bezier(0, 0, 0.39, 1) 100ms,color cubic-bezier(0, 0, 0.39, 1) 100ms,border-color cubic-bezier(0, 0, 0.39, 1) 100ms,box-shadow cubic-bezier(0.2, 0, 0.4, 0.9) 100ms;
  color: rgb(255,255,255);
  font-size: 1rem;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  min-width: 96px;
}

@media (hover: hover) {
  .emotion-11:hover {
    background-color: #004CD6;
  }
}

.emotion-11:active {
  background-color: #0025BD;
  -webkit-transform: scale(0.96);
  -moz-transform: scale(0.96);
  -ms-transform: scale(0.96);
  transform: scale(0.96);
}

.emotion-11:focus-visible {
  outline: none;
  box-shadow: 0 0 0 2px #4494EF;
}

.emotion-11 svg {
  fill: rgb(255,255,255);
}

.emotion-12 {
  min-width: 0;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <section
            class="emotion-4"
          >
            <header>
              <div
                class="emotion-5"
              >
                <div
                  class="emotion-6"
                >
                  <section
                    class="emotion-7"
                  >
                    <h1
                      class="emotion-8"
                      data-cvent-id="header-title"
                    >
                      Events+ Hubs
                    </h1>
                  </section>
                </div>
              </div>
              <section
                class="emotion-9"
              >
                <div
                  class="emotion-10"
                  data-cvent-id="header-actions"
                >
                  <button
                    class="emotion-11"
                    data-cvent-id="header-actions__create-hub"
                    type="button"
                  >
                    <span
                      class="emotion-12"
                    >
                      Create hub
                    </span>
                  </button>
                </div>
              </section>
            </header>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;
